app/models/newsletter_finder.rb

```diff
@@ -0,0 +1,27 @@
+class NewsletterFinder
```

Comment:
@kenyonj I like this idea too. Thank you. I ended up using require Enumerable. My brain hurt after it but I like the result.